src/components/ValidateConfigs.ts

```diff
@@ -0,0 +1,119 @@
+        }
+        if (!ValidateConfigs.isCompatible(props.targetGrid)) {
+            return `${widgetName}: this Mendix version is incompatible with the offline search widget`;
+        }
+        // if (!props.entity && !ValidateConfigs.isValidAttribute(props.targetGrid._datasource._entity, props)) {
+        //     // return `${widgetName}: supplied attribute name "${props.values.attribute}" does not belong to list view`;
+        //     return `${widgetName}: supplied attribute name does not belong to list view`;
+        // }
+        if (props.entity && !ValidateConfigs.itContains(props.entity, "/")) {
+            if (props.entity !== props.targetGrid._datasource._entity) {
+                return `${widgetName}: supplied entity "${props.entity}" does not belong to list view data source`;
+            }
+        }
+        if (props.entity && ValidateConfigs.itContains(props.entity, "/") && !ValidateConfigs.getRelatedEntity(props)) {
+            return `${widgetName}: supplied entity "${props.entity}" does not belong to list view data source reference`;
+        }
+        if (props.entity && ValidateConfigs.itContains(props.entity, "/")) {
+            // const entityPath = ValidateConfigs.getRelatedEntity(props);
+            // if (props.entity && !ValidateConfigs.isValidAttribute(entityPath, props)) {
+            //     // return `${widgetName}: supplied attribute name "${props.values.attribute}" does not belong to list view data source reference`;
+            //     return `${widgetName}: supplied attribute name does not belong to list view data source reference`;
+            // }
+        }
+
+        return "";
+    }
+
+    static getRelatedEntity(props: ValidateConfigProps): string {
+        if (props.targetGrid) {
+            const dataSourceEntity = window.mx.meta.getEntity(props.targetGrid._datasource._entity);
+            const referenceAttributes: string[] = dataSourceEntity.getReferenceAttributes();
+            for (const referenceAttribute of referenceAttributes) {
+                if (ValidateConfigs.itContains(props.entity, referenceAttribute)) {
+                    const selectorEntity = dataSourceEntity.getSelectorEntity(referenceAttribute);
+                    if (ValidateConfigs.itContains(props.entity, selectorEntity)) {
+                        return selectorEntity;
+                    }
+                }
+            }
+        }
+
+        return "";
+    }
+
+    // static isValidAttribute(entity: string, props: ValidateConfigProps): boolean {
+    //     if (props.targetGrid) {
+    //         const dataSourceEntity: mendix.lib.MxMetaObject = window.mx.meta.getEntity(entity);
+    //         const dataAttributes: string[] = dataSourceEntity.getAttributes();
+    //         if (ValidateConfigs.itContains(dataAttributes, props.values.)) {
+    //             return true;
+    //         }
+    //     }
+
+    //     return false;
+    // }
+
+    static isCompatible(targetGrid: ListView): boolean {
```

Comment:
Uhh? `targetGrid: ListView` make sure your naming variables consistent with the content. a grid is not a list view

Reply:
Renamed